pkg/ebpf/bpf_client.go

```diff
@@ -95,6 +95,7 @@ type BpfClient interface {
 	GetEgressProgToPodsMap() *sync.Map
 	DeletePodFromIngressProgPodCaches(podName string, podNamespace string)
 	DeletePodFromEgressProgPodCaches(podName string, podNamespace string)
+	DeletePodFromPodToAttachProbesLock(podName string, podNamespace string)
```

Comment:
What is this new Interface /method and why do we need this?

Reply:
We get to this else condition when pod is deleted. So clearing up all the internal caches for that pod to not keep the memory build up. This particular method will clean up the entry for the pod in the new map created where we are storing pod to lock mapping